validations/src/validations.rs

```diff
@@ -2090,11 +2099,11 @@ pub fn block_reward(epoch: Epoch) -> u64 {
 }
 
 /// Calculate the total amount of wits that will be rewarded to miners.
-pub fn total_block_reward() -> u64 {
+pub fn total_block_reward(initial_block_reward: u64, halving_period: u32) -> u64 {
     let mut total_reward = 0;
-    let mut base_reward = INITIAL_BLOCK_REWARD;
+    let mut base_reward = initial_block_reward;
     while base_reward != 0 {
-        total_reward += base_reward * u64::from(HALVING_PERIOD);
+        total_reward += base_reward * u64::from(halving_period);
```

Comment:
We should make sure that this function panics on overflow, just to be safe.
```suggestion
        total_reward = total_reward.checked_add(base_reward.checked_mul(u64::from(halving_period)).expect("overflow")).expect("overflow");
```